src/editor/map/context_manager.cpp

```diff
@@ -675,9 +675,11 @@ void context_manager::resize_map_dialog()
 
 void context_manager::save_map_as_dialog()
 {
+	bool first_pick = false;
 	std::string input_name = get_map_context().get_filename();
 	if(input_name.empty()) {
-		input_name = filesystem::get_legacy_editor_dir()+"/maps";
+		first_pick = true;
+		input_name = filesystem::get_current_editor_dir(editor_controller::current_addon_id_)+"/maps";
```

Comment:
What happens if there is no addon selected?

Reply:
I missed implementing that part, any idea what should it do?

Reply:
probably will need to be forward-ported.

Reply:
will save to the editor/maps directory.

Reply:
Looks like the dialog goes to `<userdata>/data/addons` then (that's what happens in master anyway). It should go to `<userdata>/editor/maps` as before.

Reply:
Fixed in 675e7620e67fa5df1ac09ae784c1fc070b128d99